prometheus-pushgateway/prometheus-pushgateway.spec

```diff
@@ -0,0 +1,209 @@
+rather a metrics cache. It does not have statsd-like semantics. The metrics
+pushed are exactly the same as you would present for scraping in a permanently
+running program.
+
+For machine-level metrics, the textfile collector of the Node exporter is
+usually more appropriate. The Pushgateway is intended for service-level metrics.
+
+The Pushgateway is not an event store. While you can use Prometheus as a data
+source for Grafana annotations, tracking something like release events has to
+happen with some event-logging framework.
+
+################################################################################
+
+%prep
+%setup -q -n %{pkg_name}-%{version}
+
+mkdir -p .src/github.com/prometheus/%{pkg_name}
+mv * .src/github.com/prometheus/%{pkg_name}/
+mv .promu.yml .src/github.com/prometheus/%{pkg_name}/
+mv .src src
+
+%build
+export GOPATH=$(pwd)
+
+pushd src/github.com/prometheus/%{pkg_name}
+    %{__make} %{?_smp_mflags} build
+popd
+
+%install
+rm -rf %{buildroot}
+
+install -dm 0755 %{buildroot}%{_bindir}
+install -dm 0755 %{buildroot}%{_sysconfdir}/%{name}
+install -dm 0755 %{buildroot}%{_sysconfdir}/logrotate.d
+install -dm 0755 %{buildroot}%{_sysconfdir}/sysconfig
+install -dm 0755 %{buildroot}%{_logdir}/%{name}
+
+%if 0%{?rhel} >= 7
+    install -dm 0755 %{buildroot}%{_unitdir}
+%else
+    install -dm 0755 %{buildroot}%{_initrddir}
+%endif
+
+install -pm 0755 src/github.com/prometheus/%{pkg_name}/%{pkg_name} %{buildroot}%{_bindir}/%{name}
+install -pm 0644 %{SOURCE3} %{buildroot}%{_sysconfdir}/sysconfig/%{name}
+install -pm 0644 %{SOURCE4} %{buildroot}%{_sysconfdir}/logrotate.d/%{name}
+
+%if 0%{?rhel} >= 7
+    install -pm 0644 %{SOURCE1} %{buildroot}%{_unitdir}/%{name}.service
+%else
+    install -pm 0755 %{SOURCE2} %{buildroot}%{_initrddir}/%{name}
+%endif
+
+%clean
+rm -rf %{buildroot}
+
+################################################################################
+
+%pre
+%{__getent} group %{service_group} >/dev/null || %{__groupadd} -r %{service_group}
+%{__getent} passwd %{service_user} >/dev/null || %{__useradd} -r \
+    -g %{service_group} -d %{_sharedstatedir}/%{name} \
+    -s /sbin/nologin %{service_user}
+
+%post
+if [[ $1 -eq 1 ]] ; then
+%if 0%{?rhel} >= 7
+  %{__systemctl} enable %{name}.service &>/dev/null || :
+%else
+  %{__chkconfig} --add %{name} &>/dev/null || :
+%endif
+fi
+
+%preun
+if [[ $1 -eq 0 ]] ; then
+%if 0%{?rhel} >= 7
+  %{__systemctl} --no-reload disable %{name}.service &>/dev/null || :
+  %{__systemctl} stop %{name}.service &>/dev/null || :
+%else
+  %{__service} %{name} stop &>/dev/null || :
+  %{__chkconfig} --del %{name} &>/dev/null || :
+%endif
+fi
+
+%postun
+%if 0%{?rhel} >= 7
+if [[ $1 -ge 1 ]] ; then
+  %{__systemctl} daemon-reload &>/dev/null || :
+fi
+%endif
+
+################################################################################
+
+%files
+%defattr(-,root,root,-)
+%{_bindir}/%{name}
+%if 0%{?rhel} >= 7
+    %{_unitdir}/%{name}.service
```

Comment:
Don't use prefix spaces in `%files` section.

Reply:
Fixed (b11385b0389466a952508b7cf1e0c755d1f8add7).